Source/Eto.Wpf/Forms/Cells/ProgressCellHandler.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Eto.Forms;
+using swc = System.Windows.Controls;
+using swd = System.Windows.Data;
+using sw = System.Windows;
+using Test;
+
+namespace Eto.Wpf.Forms.Cells
+{
+	public class ProgressCellHandler : CellHandler<DataGridProgressBarColumn, ProgressCell, ProgressCell.ICallback>, ProgressCell.IHandler
+	{
+		public int GetValue(object dataItem)
+		{
+			if (Widget.Binding != null)
+				return (int)Widget.Binding.GetValue(dataItem);
+			return 0;
+		}
+
+		public void SetValue(object dataItem, int value)
+		{
+			if (Widget.Binding != null)
+				Widget.Binding.SetValue(dataItem, value);
+		}
+
+		class Column : DataGridProgressBarColumn
+		{
+			public ProgressCellHandler Handler { get; set; }
+
+			protected override sw.FrameworkElement GenerateElement(swc.DataGridCell cell, object dataItem)
+			{
+				var element = base.GenerateElement(cell, dataItem);
+				element.DataContextChanged += (sender, e) =>
+				{
+					var grid = sender as swc.Grid;
+
+					swc.ProgressBar progressBar = null;
+					swc.TextBlock textBlock = null;
+
+					foreach (var child in grid.Children)
+						if (progressBar != null && textBlock != null)
+							break;
+						else if (child is swc.ProgressBar)
+							progressBar = child as swc.ProgressBar;
+						else if (child is swc.TextBlock)
+							textBlock = child as swc.TextBlock;
+
+					progressBar.Value = Handler.GetValue(grid.DataContext);
+					textBlock.Text = ((int)progressBar.Value) + "%";
+
+					Handler.FormatCell(grid, cell, dataItem);
+				};
+				return Handler.SetupCell(element);
+			}
+		}
+
+		public ProgressCellHandler()
+		{
+			Control = new Column { Handler = this };
+		}
+	}
+}
+
+
+namespace Test

```

Comment:
Oops, didn't see this before.. Could you move the class out of the Test namespace?